File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_092646/typescript/002_word_frequency/solution.ts
import { readFileSync } from 'fs';

const filePath = process.argv.length > 2 ? process.argv[2] : 'input/text.txt';

try {
  const text = readFileSync(filePath, { encoding: 'utf-8' });
  
  // Convert to lowercase and extract words consisting of only letters
  const words = text.toLowerCase().match(/[a-z]+/g);
  
  const freqMap = new Map<string, number>();
  
  if (words) {
    for (const word of words) {
      const currentCount = freqMap.get(word) || 0;
      freqMap.set(word, currentCount + 1);
    }
  }
  
  // Convert entries to array
  const result: [string, number][] = Array.from(freqMap.entries());
  
  // Sort by count descending, then word ascending
  result.sort((a, b) => {
    if (a[1] !== b[1]) {
      return b[1] - a[1];
    } else {
      return a[0].localeCompare(b[0]);
    }
  });
  
  for (const [word, count] of result) {
    console.log(`${word}: ${count}`);
  }
} catch (err) {
  // Do not output anything else
  process.exit(1);
}

File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_092646/python/002_word_frequency/solution.py
import re
from collections import Counter

def count_word_frequencies(filename):
    # Read and process text
    with open(filename, 'r', encoding='utf-8') as f:
        text = f.read()
    
    # Extract alphabetic words (case-insensitive via regex)
    words = re.findall(r'[a-zA-Z]+', text)
    
    # Count lowercase occurrences
    word_counts = Counter(word.lower() for word in words)
    
    # Sort by count descending, then alphabetically ascending
    sorted_words = sorted(word_counts.items(), key=lambda x: (-x[1], x[0]))
    
    # Output results to stdout
    for word, count in sorted_words:
        print(f"{word}: {count}")

if __name__ == "__main__":
    count_word_frequencies("input/text.txt")